python/ray/train/v2/tests/test_controller.py

```diff
@@ -316,6 +329,35 @@ def before_controller_shutdown(self):
     assert callback.shutdown_called
 
 
+@pytest.mark.asyncio
+async def test_controller_abort(tmp_path):
+    class AbortCallback(ControllerCallback):
+        def before_controller_abort(
+            self, worker_group_context: Optional[WorkerGroupContext]
+        ):
+            tmp_path.joinpath("before_controller_abort").touch()
+
+    callback = AbortCallback()
+
+    scaling_policy = MockScalingPolicy(scaling_config=ScalingConfig())
+    failure_policy = MockFailurePolicy(failure_config=None)
+    train_run_context = TrainRunContext(run_config=RunConfig())
+
+    train_controller_cls = ray.remote(TrainController)
+
+    controller = train_controller_cls.remote(
+        train_fn_ref=DummyObjectRefWrapper(lambda: None),
+        train_run_context=train_run_context,
+        scaling_policy=scaling_policy,
+        failure_policy=failure_policy,
+        callbacks=[callback],
+    )
```

Comment:
I don't want this test to need to create the actor itself -- we can just test the TrainController logic with the "local version." Doing this keeps the test running fast and makes it simpler (actually testing w/ Ray actor doesn't really add much to these unit tests).

Ex: we can mock the exit_actor call to just check that it's happening, and we should assert that the controller is getting set to the ABORTED state as expected.

Reply:
This seems a bit wonky to me because `abort` will only ever get called as an actor method (I can also add a comment clarifying that). In addition making actors in unit tests seems relatively common and not that slow e.g. `RAY_TRAIN_V2_ENABLED=1 pytest -vs python/ray/train/v2/tests/test_controller.py::test_controller_abort` with the actor took me `4.1s` and in local mode took me `3.84` and failed with `exit_actor API is called on a non-actor worker`. Lmk what you think.

Reply:
The other tests in this test suite are also only called as actor methods, but calling the actor methods locally makes it easier to make assertions, since running it as an actor puts all the relevant state in a different process that is hard to access.

I do agree the runtime is probably not a big deal -- the only case it becomes bad is if you have a lot of tests that require `ray.init()` without sharing the fixture (by setting `pytest.fixture(scope="module")`), then it can be like 1s+ of overhead per test.

Reply:
I think my idea is to avoid the data parallel trainer style tests when possible: otherwise we might end up in a situation where everything is basically an end to end test (like the Tune tests https://github.com/ray-project/ray/blob/master/python/ray/tune/tests/test_tune_restore.py#L234)

Reply:
sgtm - done.